InterGalacticCurrencyConvertor/src/main/java/DataProcessor.java

```diff
@@ -1,80 +1,73 @@
 
-	private void loadClasses(String inPackageName, Class<?> inInterfaceName) {
-		try {
-			FindPackageClasses.loadClassesImpInterface(inPackageName,
-			                                           inInterfaceName);
-		} catch (ClassNotFoundException | IOException e) {
-			System.out.println(e);
-			System.exit(-1);
-		}
-	}
-	public static void registerQueryObjects(Query inQuery) {
-	  Objects.requireNonNull(inQuery, "Invalid Query subclass");
-		mQueryObjList.add(inQuery);
-	}
-	public static void registerStatementObjects(Statement inStatement) {
-	  Objects.requireNonNull(inStatement, "Invalid Statement subclass");
-		mStatementObjList.add(inStatement);
+	static {
+	  // Needs to be updated when new Processor classes are added
+	  mQueryProcessorList = new ArrayList<>();
+	  mQueryProcessorList.add(new InterGalacticToDecimalQueryProcessor());
+	  mQueryProcessorList.add(new MaterialCostQueryProcessor());
+	  
+	  mStatementProcessorList = new ArrayList<>();
+	  mStatementProcessorList.add(new InterGalacticToRomanMapStatementProcessor());
+	  mStatementProcessorList.add(new MaterialCostStatementProcessor());
 	}
-	public DataProcessor() {
-		loadClasses(Statement.class.getPackage().getName(), Statement.class);
-		loadClasses(Query.class.getPackage().getName(), Query.class);
+	private DataProcessor() {}
+	public static DataProcessor getInstance() {
+	  return mInstance;
 	}
-	private DataProcessResult processQuery(String [] inTokens) {
-		DataProcessResult vResult = DataProcessResult.EMPTY_RESULT;
-		for (Query query: mQueryObjList) {
-			vResult = query.processQuery(inTokens);
-			if (vResult.hasSucceeded()) {
+	private DataProcessResult processQuery(ProcessorData processorData) {
+		DataProcessResult result = DataProcessResult.EMPTY_RESULT;
+		for (QueryProcessor queryProcessor: mQueryProcessorList) {
+			result = queryProcessor.processQuery(processorData);
+			if (result.isSuccess()) {
 				break;
 			}
 		}
-		return vResult;
+		return result;
 	}
-	private DataProcessResult processStatement(String [] inTokens) {
-		DataProcessResult vResult = DataProcessResult.EMPTY_RESULT;
-		for (Statement stm: mStatementObjList) {
-			vResult = stm.processStatement(inTokens);
-			if (vResult.hasSucceeded()) {
+	private DataProcessResult processStatement(ProcessorData processorData) {
+		DataProcessResult result = DataProcessResult.EMPTY_RESULT;
+		for (StatementProcessor stmProcessor: mStatementProcessorList) {
+			result = stmProcessor.processStatement(processorData);
+			if (result.isSuccess()) {
 				break;
 			}
 		}
-		return vResult;
+		return result;
 	}
-	public DataProcessResult processData(String inLine) {
-	  Objects.requireNonNull(inLine, "Invalid Data input");
-		DataProcessResult vResult = DataProcessResult.EMPTY_RESULT;
-		inLine = inLine.toLowerCase();
-		if (inLine.equals("exit")) {
-			DataReader.setExit(true);
-			return vResult;
+	public DataProcessResult processData(String data) {
```

Comment:
Consider trimming whitespace at the beginning, end and middle (if it makes the following regexes simpler.

Reply:
done in ProcessorData class